### PR TITLE
Use pytest functions and fixtures for integration tests

### PR DESCRIPTION
**Pull request by sopelj**
Depends-on: https://github.com/wazo-platform/wazo-test-helpers/pull/89

### CODE REVIEW

**Comment by sopelj** [comment]
recheck (CI stopped the build)

File: integration_tests/suite/helpers/base.py [hunk 1]
```diff
@@ -12,48 +11,58 @@
...
         return CalldMockClient('127.0.0.1', cls.service_port('9500', 'calld'))
 
-    @classmethod
-    def make_database(cls):
+    @cached_class_property
+    def db(cls) -> DbHelper | WrongClient:
         try:
             port = cls.service_port(5432, 'postgres')
         except (NoSuchService, NoSuchPort):
             return WrongClient('postgres')
```
Comment (fblackburn1) [inline integration_tests/suite/helpers/base.py:63]: Does the `WrongClient` still make sense?
From what I understand, the `cached_class_property` and the `WrongClient` play the same role, i.e. make the client lazy and fail only on the first usage

Comment (sopelj) [reply integration_tests/suite/helpers/base.py:63]: Oh OK, I wasn't sure the overlap was exactly the same, but no, it doesn't really make sense unless it is used elsewhere and it's too complicated to change.

Comment (fblackburn1) [reply integration_tests/suite/helpers/base.py:63]: Not used elsewhere, you can safely remove it :crossed_fingers:

**Review by sduthil — APPROVED** [review]

**Comment by sduthil** [comment]
@sopelj do you know if the test failure is linked to the PR?

**Comment by sopelj** [comment]
> @sopelj do you know if the test failure is linked to the PR?

I don't think it is. I have the same problem on master. I'll rebase with latest changes and see.

**Comment by sopelj** [comment]
> > @sopelj do you know if the test failure is linked to the PR?
> 
> I don't think it is. I have the same problem on master. I'll rebase with latest changes and see.

@sduthil No, the issue was in master and has since been resolved. Rebasing seems to have resolved it.

**Review by sduthil — APPROVED** [review]

**Pull request merged**